src/v/config/base_property.h

```diff
@@ -77,6 +83,23 @@ class base_property {
     virtual void reset() = 0;
     virtual bool is_default() const = 0;
 
+    /**
+     * Helper for logging string-ized values of a property, e.g.
+     * while processing an API request or loading from file, before
+     * the property itself is initialized.
+     *
+     * Use this to ensure that any logged values are properly
+     * redacted if secret.
+     */
+    template<typename U>
+    std::string_view format_raw(U const& in) {
+        if (is_secret() && !in.empty()) {
+            return secret_placeholder;
+        } else {
+            return in;
```

Comment:
will this leak my secure password: `""` ?

Reply:
I mean it will now that you've told me that's your password :-D